joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/datastructures/CSharpScope.scala

```diff
@@ -47,7 +48,17 @@ class CSharpScope(typeMap: TypeMap) extends Scope[String, DeclarationNew, ScopeT
     .exists(x => x.scopeNode.isInstanceOf[MethodScope] || x.scopeNode.isInstanceOf[TypeLikeScope])
 
   def tryResolveTypeReference(typeName: String): Option[String] = {
-    typesInScope.find(_.name.endsWith(typeName)).flatMap(typeMap.namespaceFor).map(n => s"$n.$typeName")
+    typesInScope
+      .find(_.name.endsWith(typeName))
+      .flatMap(typeMap.namespaceFor)
+      .map(n => {
+        // To avoid recursive type prefixing on assignment calls.
+        if (typeName.startsWith(n)) {
+          return Some(typeName)
```

Comment:
It's generally better to use `Option` instead of `Some` directly, but I wont block the PR for that now

Reply:
^ Reason being, `Some(null) != Option(null)`